### PR TITLE
Added PostitiveIntegerRule for validating ids

### PR DESCRIPTION
**Pull request by eugenebond**
@ingluisjimenez @mikesherov

### CODE REVIEW

**Comment by ingluisjimenez** [comment]
:+1:

File: tests/Validation/Rules/PostitiveIntegerRuleTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        [ "1", true ],
+        [ "-", false ],
+        [ 1.01, false ],
+        [ -1.01, false ],
+        [ "1.0", false ],
+        [ "-1.0", false ],
+        [ 1.43e26, false ],
+        [ "1.43e26", false ],
+        [ 'false', false ],
+        [ 'true', false ]
```
Comment (mikesherov) [inline tests/Validation/Rules/PostitiveIntegerRuleTest.php]: missing `false`, `true`, `1.0e26` test cases.

**Comment by mikesherov** [comment]
:+1:

File: tests/Validation/Rules/PostitiveIntegerRuleTest.php (hunk 1 above)
Comment (eugenebond) [inline tests/Validation/Rules/PostitiveIntegerRuleTest.php]: I can add true and false but those produce somewhat unexpected results - the input value gets converted to string so `true` will become `"1"` and will pass validation.

Comment (eugenebond) [inline tests/Validation/Rules/PostitiveIntegerRuleTest.php]: perhaps I should try implementing it as a non-regex rule

File: src/Validation/Rules/PostitiveIntegerRule.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * will statically call a parameterized class, method name as the validation
+ */
+class PostitiveIntegerRule extends CallbackRuleAbstract {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct() {
+
+    $closure = ( function( $data ) {
```
Comment (mikesherov) [inline src/Validation/Rules/PostitiveIntegerRule.php]: unnecessary temp var. Creates unnecessary obfuscation.

Comment (mikesherov) [inline src/Validation/Rules/PostitiveIntegerRule.php]: even if the variable is intending to clarify purpose, `$closure` is too literal of a variable name.

File: tests/Validation/Rules/PostitiveIntegerRuleTest.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        [ -1.01, false ],
+        [ "1.0", false ],
+        [ "-1.0", false ],
+        [ 1.43e26, false ],
+        [ 1.0e26, false ],
+        [ "1.43e26", false ],
+        [ 'false', false ],
+        [ 'true', false ],
+        [ false, false ],
+        [ true, false ],
```
Comment (mikesherov) [inline tests/Validation/Rules/PostitiveIntegerRuleTest.php]: while we're at it, what about `null`?

Comment (eugenebond) [inline tests/Validation/Rules/PostitiveIntegerRuleTest.php]: good idea; added

**Comment by ingluisjimenez** [comment]
:+1:

**Pull request merged**